main/src/com/google/refine/util/DetectLanguageUtils.java

```diff
@@ -15,11 +14,18 @@
 import java.io.IOException;
 import java.util.List;
 
-public class LangDetectUtils {
+public class DetectLanguageUtils {
+    private static List<LanguageProfile> languageProfiles;
 
-    public static Optional<LdLocale> detect(String text) throws IOException {
-        List<LanguageProfile> languageProfiles = new LanguageProfileReader().readAllBuiltIn();
+    static {
+        try {
+            languageProfiles = new LanguageProfileReader().readAllBuiltIn();
```

Comment:
Having this in a static block will indeed load the language profiles only once, which is better, but I think it would be even nicer if that could only happen the first time the GREL function is used.

Reply:
🤔 ah! I get it so clearly now!
I'll work something out 😃